psydac/linalg/block.py

```diff
@@ -1255,13 +1255,13 @@ def set_backend(self, backend):
 
         if interface:
             def func(blocks, v, out, **args):
-                    vs   = [vi._interface_data[d_axis, d_ext] for vi in v.blocks] if isinstance(v, BlockVector) else v._data
-                    outs = [outi._data for outi in out.blocks] if isinstance(out, BlockVector) else out._data
+                    vs   = [vi._interface_data[d_axis, d_ext] for vi in v.blocks] if isinstance(v, BlockVector) else [v._data]
```

Comment:
@yguclu, @vcarlier, I have a hard time recreating the problem, could you help me out?
So far I've tried
```
# V is a StencilVectorSpace
# and S is a StencilMatrix from V -> V
U = BlockVectorSpace(V, V)

# 2 different block arrangements for both a BlockLinearOperator from U -> V as well as V -> U
dict_blocks1 = {(0,0):S, (0,1):S}
dict_blocks2 = {(0,0):S, (1,0):S}
B1 = BlockLinearOperator(U, V, blocks=dict_blocks1)
B2 = BlockLinearOperator(V, U, blocks=dict_blocks2)

# s is a StencilVector that belongs to the Space V
b = BlockVector(U, (s, s))

y1 = B1 @ b
y2 = B2 @ s
```
Both LinearOperators, although either from BlockVectorSpace/StencilVectorSpace to StencilVectorSpace/BlockVectorSpace, evaluate just the way they should.